src/Pages/AuthenticationPage/AuthenticationPage.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import { AppDispatch } from '../../store';
+import { setAuth } from '../../features/auth/auth';
+
+export default function AuthenticationPage() {
```

Comment:
I think all of this logic can be encapsulated inside a function called in `App.tsx` on mount, and, a custom hook let's called `function useIsAuthenticated(): boolean`, or a reducer selector, can be used inside a `PrivateRoute` to redirect the user to a login page if authentication fail.

You have succeeded calling the endpoint using `fetch`, I suggest using `Axios`, please read it's documentation before you start using it, as its flexible to handle JWT authentication, and it's used inside schedex :P.

Reply:
Mooved the logic to a custom hook using axios only need confirmation if I need to move the hook call to main , please the the comment above.